main.tf

```diff
@@ -542,28 +532,28 @@ resource "aws_autoscaling_policy" "this" {
       scheduling_buffer_time       = lookup(predictive_scaling_configuration.value, "scheduling_buffer_time", null)
 
       dynamic "metric_specification" {
-        for_each = lookup(predictive_scaling_configuration.value, "metric_specification", [])
+        for_each = can(predictive_scaling_configuration.value.metric_specification.target_value) ? [predictive_scaling_configuration.value.metric_specification] : []
```

Comment:
with the bug fix on line 560, I had to make some adjustments to get this to work as intended and without errors